internal/controller/nats/controller_test.go

```diff
@@ -82,13 +83,15 @@ func Test_initNATSInstance(t *testing.T) {
 			// define mocks behaviour
 			testEnv.kubeClient.On("DestinationRuleCRDExists",
 				mock.Anything).Return(tc.wantIstioEnabled, nil)
-			if tc.wantSecretExists {
-				sampleSecret := testutils.NewSecret()
+			if tc.wantRotatePassword {
```

Comment:
rotating a secret usually means that you replace an existing secret with a new secret